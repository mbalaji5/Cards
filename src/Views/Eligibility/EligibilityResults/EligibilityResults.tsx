import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Error from "../../Error/Error";

const ResultsWrapper = styled.div`
  flex: 1 1 auto;
  padding-top: 48px;
  justify-content: center;
  margin: 0 -8px;
  display: flex;
  flex-wrap: wrap;
`;

interface propType {
  eligibilityResult: any;
}

const EligibilityResults = ({ eligibilityResult }: propType) => {
  const [errorMessage, setErrorMessage] = useState("");
  const [eligibleProduct, setEligibleProduct] = useState("");

  useEffect(() => {
    if (eligibilityResult && eligibilityResult.cardType) {
      let eligibileCard = "";
      eligibilityResult.cardType.forEach((card: string) => {
        if (eligibilityResult.eligibility[card].isEligible)
          eligibileCard = eligibileCard ? `${eligibileCard} and ${card}` : card;
      });

      const eligibleProductLabel = eligibileCard
        ? `Congratulations you are eligible for ${eligibileCard} credit card product.`
        : `Sorry! You are not eligible for neither C1 nor C2.`;

      setEligibleProduct(eligibleProductLabel);
    } else if (eligibilityResult && eligibilityResult.error) {
      setErrorMessage(
        "Sorry! Currently we are unable to precess your request."
      );
    }
  }, [eligibilityResult]);

  return (
    <ResultsWrapper>
      {eligibleProduct && <label>{eligibleProduct}</label>}
      {errorMessage && <Error errorMessage={errorMessage} />}
    </ResultsWrapper>
  );
};

EligibilityResults.defaultProps = { eligibilityResult: {} };

export default EligibilityResults;
