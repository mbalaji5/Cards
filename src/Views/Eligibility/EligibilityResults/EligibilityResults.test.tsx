import { render } from "@testing-library/react";
import React from "react";
import EligibilityResults from "./EligibilityResults";

describe("Should Validate EligibilityResults Component", () => {
  it("Should validate eligibilityResult With Valid C1 and C2", () => {
    const eligibilityResult = {
      cardType: ["C1", "C2"],
      eligibility: {
        C1: { isEligible: true },
        C2: { isEligible: true },
      },
    };
    const { container, baseElement } = render(
      <EligibilityResults eligibilityResult={eligibilityResult} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(container.textContent).toBe(
      "Congratulations you are eligible for C1 and C2 credit card product."
    );
  });
  it("Should validate eligibilityResult With Valid C1 only", () => {
    const eligibilityResult = {
      cardType: ["C1", "C2"],
      eligibility: {
        C1: { isEligible: true },
        C2: { isEligible: false },
      },
    };
    const { container, baseElement } = render(
      <EligibilityResults eligibilityResult={eligibilityResult} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(container.textContent).toBe(
      "Congratulations you are eligible for C1 credit card product."
    );
  });
  it("Should validate eligibilityResult With Valid C2 only", () => {
    const eligibilityResult = {
      cardType: ["C1", "C2"],
      eligibility: {
        C1: { isEligible: false },
        C2: { isEligible: true },
      },
    };

    const { container, baseElement } = render(
      <EligibilityResults eligibilityResult={eligibilityResult} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(container.textContent).toBe(
      "Congratulations you are eligible for C2 credit card product."
    );
  });
  it("Should validate eligibilityResult With in Valid C1 and C2", () => {
    const eligibilityResult = {
      cardType: ["C1", "C2"],
      eligibility: {
        C1: { isEligible: false },
        C2: { isEligible: false },
      },
    };
    const { container, baseElement } = render(
      <EligibilityResults eligibilityResult={eligibilityResult} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(container.textContent).toBe(
      "Sorry! You are not eligible for neither C1 nor C2."
    );
  });
  it("Should validate error response from downstream", () => {
    const eligibilityResult = {
      error: { message: "Internal server error." },
    };
    const { container, baseElement } = render(
      <EligibilityResults eligibilityResult={eligibilityResult} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(container.textContent).toBe(
      "Sorry! Currently we are unable to precess your request."
    );
  });
});
