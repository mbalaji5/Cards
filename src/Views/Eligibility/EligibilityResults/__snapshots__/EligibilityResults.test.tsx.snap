// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Validate EligibilityResults Component Should validate eligibilityResult With Valid C1 and C2 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB eAvOvM"
    >
      <label>
        Congratulations you are eligible for C1 and C2 credit card product.
      </label>
      
    </div>
  </div>
</body>
`;

exports[`Should Validate EligibilityResults Component Should validate eligibilityResult With Valid C1 only 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB eAvOvM"
    >
      <label>
        Congratulations you are eligible for C1 credit card product.
      </label>
      
    </div>
  </div>
</body>
`;

exports[`Should Validate EligibilityResults Component Should validate eligibilityResult With Valid C2 only 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB eAvOvM"
    >
      <label>
        Congratulations you are eligible for C2 credit card product.
      </label>
      
    </div>
  </div>
</body>
`;

exports[`Should Validate EligibilityResults Component Should validate eligibilityResult With in Valid C1 and C2 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB eAvOvM"
    >
      <label>
        Sorry! You are not eligible for neither C1 nor C2.
      </label>
      
    </div>
  </div>
</body>
`;

exports[`Should Validate EligibilityResults Component Should validate error response from downstream 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBQB eAvOvM"
    >
      
      <label
        id="error"
      >
        Sorry! Currently we are unable to precess your request.
      </label>
    </div>
  </div>
</body>
`;
